app/Main.hs

```diff
@@ -42,16 +42,21 @@ getPlayer playerNumber gameState = case playerNumber of
                                         Player1 -> player1 gameState
                                         Player2 -> player2 gameState 
 
-flagPrompt :: IO Int
-flagPrompt = do
+chooseFlag :: PlayerNumber -> GameState -> IO Flag
+chooseFlag playerNumber gameState = do
     input <- ask "Please pick a flag"
-    let result = reads input :: [(Int, String)]
-    case result of 
-        (first:rest) -> return $ fst first
-        _ -> flagPrompt
+    case input of
```

Comment:
What do you think about pulling this into a function that pattern matches and returns a Maybe Flag

```haskell
flagFromString :: String -> Maybe Flag
```